### DISCUSSION TITLE
Compartilhe sua experiência que você teve em suas entrevistas como Programador

### DISCUSSION THREAD

**Discussion by evertonProgramadorCriativo** [frontendbr/forum#1637] [Discussão]
Conte-me quais as principias perguntas e exercícios que os recrutadores fizeram com voce e compartilhe sua experiencia aqui !!!

**Comment by eliseumds** [comment]
Em 2011, quando eu tinha 16 anos, ainda morando em Praia Grande - SP, fui contatado pelo dono de uma agência que produzia sites para imobiliárias, um mercado que estava explodindo na cidade. Ele me encontrou no fórum iMasters onde eu curtia responder a galera todo dia. Cheguei no escritório e me pediram pra codar um layout que eles tinham (HTML, CSS e um pouco de JS pra dropdown menus e tal) e felizmente terminei em pouco tempo. Não havia muito talento na cidade, e eles me contrataram no mesmo dia. Eu não precisava trabalhar na época então não fiquei nervoso na entrevista.

Em 2012 eu precisava me mudar pra São Paulo pra iniciar o curso universitário e, no mesmo dia da matrícula, uma empresa que controlava alguns sites gigantescos de e-commerce (esqueci o nome) e a Vitalbox (startup de bioinformática) me chamaram pra entrevistas:

1. Na primeira, a entrevista foi somente oral, num prédio gigantesco, a vaga era pra front-end também, e acabei passando no mesmo dia com proposta de salário, mas eu estava indo pra São Paulo sozinho, precisaria alugar um lugar pra morar, e com aquele valor eu não iria conseguir. Eu entendo, eu tinha 17 anos, sabia que seria tenso.
1. Depois fui pra entrevista com a Vitalbox numa padaria, e confesso que eu estava bem nervoso pq era exatamente o que eu estava buscando: um time pirado em tecnologia num ambiente descontraído. Depois que me pediram pra escrever FizzBuzz eu pensei "Puta merda, sério?". Eu disse que FizzBuzz eu conseguiria resolver em uma linha e pedi algo mais desafiador. Por um lado, consegui demonstrar confiança e fiquei mais relaxado, por outro, eu não faria isso de novo pq hoje entendo que um simples FizzBuzz pode desencadear uma porrada de questões diferentes, e úteis. Depois disso implementamos tic-tac-toe juntos com JS puro. Pra ser honesto, não acho que implementei a coisa mais linda do mundo, mas acho que a nossa conversa foi mais importante pra eles, e ao demonstrar confiança e entusiasmo eu definitivamente ganhei uns pontos. Me ofereceram o dobro do salário da outra empresa, mais um apartamento a 10 minutos a pé do escritório. Eu achei que estavam brincando com a minha cara, e aceitei depois de alguns segundos de silêncio.

Em 2014, minha namorada me contou que havia passado no vestibular da Universidade de Coimbra e teria que se mudar em um mês. Eu fiquei abalado mas também usei isso como motivação pra tentar algo fora do país. Como os salários em Portugal eram muito baixos, eu considerei principalmente Londres, Amsterdã e Berlim, que ficam a curtos voos de lá. Enviei muitos e-mails e consegui algumas entrevistas via Skype, todas em inglês, então eu fiquei mais nervoso que nas entrevistas anteriores. Acho que passei na maioria delas, mas pela dificuldade de conseguir vistos, eu acabei me inclinando mais à Alemanha. As entrevistas, agora que eu já tinha mais experiência, foram menos focadas em implementar algoritmos específicos, e mais em falar sobre frameworks, design patterns e soft skills. Na mesma época eu também entrevistei com outras empresas de São Paulo e não vi qualquer diferença entre brasileiros e estrangeiros, além da língua.

No final acabei me identificando mais com a startup [42reports de Berlim](https://42reports.com), um projeto muito bacana e com tecnologia avançada (recebíamos mais de 1 bilhão de sinais de roteadores 3G, Bluetooth e câmeras 3D ao redor da Europa, por minuto). Eu já estava trabalhando com ReactJS na Vitalbox, bem early adopter, e o frontend dev da 42reports queria aprendê-lo e substituir a impementação AngularJS, então a entrevista foi focada nestes frameworks e em HTML5 Canvas. Ao final da entrevista eles me perguntaram qual era minha pretensão salarial e acho que nessa hora eu ganhei uns pontos: como eu estava com salários de Londres e Amsterdã na mente, eu usei o mesmo valor nesta entrevista, mas eu não tinha muita noção que o custo de vida em Berlim era muito menor e consequentemente salários também. Eu pedi 70k euros/ano mais relocation support, o que era um valor absurdo, principalmente para alguém sem diploma e com 19 anos como eu, mas acho que com isso eu consegui demonstrar confiança. No final fechamos por 55k e eles me levaram pra Berlim por um mês pra conhecer o time e depois assinar um contrato de vdd, se tudo desse certo. Cheguei lá e o frontend dev deles pediu demissão uma semana depois, então eles não tinham muita escolha a não ser me contratar, mas eu acabei me lascando pq tive que usar o lixo do AngularJS pq migrar tudo pra React sozinho seria inviável. Apesar disso, foram dois anos incríveis de aprendizado e que me colocaram num patamar profissional muito mais elevado.

Em 2016 eu estava fuçando no canal [Reactiflux no Discord](https://www.reactiflux.com/) e vi um job post da empresa [ProductReview.com.au](https://www.productreview.com.au) baseada em Sydney. Na época eu estava interessado em Domain Driven Design e o tech lead me impressionou muito. Era um código muito lindo e claro para uma empresa tão pequena. Eles estavam meio desesperados depois de más experiências usando freelancers de outros países e fizemos uma entrevista via Skype no dia seguinte, e mais uma vez eu já os cortei logo no começo pq começaram a fazer umas perguntas esquisitas sobre JavaScript como `[] + {}` e `new Array(3) === ",,"`. Hoisting, scope, binding e Promises eu aceito, mas essas perguntas inúteis me deixam puto. Eu não sabia algumas das respostas, e com essa cortada eu ganhei pontos, não as respondi, e partimos para uma entrevista mais decente, uma semana depois. Como o time era bem pequeno (e ainda é), eles esperavam mais de um desenvolvedor, alguém com conhecimento mais abrangente que estivesse confortável com a stack inteira, então respondi perguntas sobre arquitetura, SQL, SOLID, OOP, testes, todas high-level. Se a entrevista é bem feita e fluída, nem de white board você precisa. Eu só codei nas minhas primeiras entrevistas até 2012, depois foi só oral.

Antes dessa entrevista pra ProductReview.com.au, eu entrevistei com uma empresa de San Fran só por diversão e foi uma desgraça, pra mim. Me pediram pra implementar uns algoritmos recursivos em árvores, algo que eu já tinha feito no curso técnico e na universidade e conseguiria fazer hoje, mas falhei. Eu não entendo o ponto de entrevistas assim para empresas que criam produtos high-level. Eu admito que fiquei um pouco bolado pq sou exigente comigo mesmo, mas vida que segue. Por outro lado, eu entrevistei um frontend em Berlim e fiz umas perguntas ridículas sobre umas APIs que ninguém usava, e me arrependo. Foi minha pior execução como entrevistador, eu simplesmente estava querendo parecer inteligentão, e acabei parecendo um robô bobão. Coisas que a gente aprende.

No começo deste ano (2019), eu entrevistei com uma grande empresa de investimentos em São Paulo e novamente não houve qualquer white board interview. Algumas perguntas de alto nível aqui, outras de baixo nível ali, e isso é o suficiente pra avaliar as skills técnicas de um candidato, mas o que conta mesmo quando você é mais experiente é sua capacidade de comunicação, de delegar tarefas, de evitar problemas futuros pq já viu merda acontecer. Essas entrevistas são muito mais complexas e subjetivas, então simpatia e eloquência pesam bastante, mas principalmente a humildade de dizer que não conhece algum tópico. Isso demonstra seriedade, maturidade, exatamente o que empresas grandes querem, pq elas não querem perder tempo. No final eu acabei não seguindo com o processo por causa da baixa qualidade de vida em São Paulo.

Em resumo, eu penso que minhas habilidades técnicas e a falta de concorrência me ajudaram no começo da minha carreira; eu era um rato da web, fuçava em tudo que era possível, sabia como cada CSS rule iria funcionar em cada browser, mas eu era um macro que apenas executava as ordens, barata tonta. Hoje eu sou muito mais calmo e foco no problema por completo, tento ser empático e pensar como o consumidor do nosso produto e tento pensar como meu team lead. Eu cometi muitos erros por imaturidade mas eles foram fundamentais pro meu desenvolvimento.
